#119 Native Sublinks API

Pdzly opened the pull request:
Resolves #119 

Implements #406 #425

jgrim committed:
#119 Created base controllers for native API

jgrim committed:
Merge branch 'main' into issue/119-native-api

Pdzly committed:
Minor fix addition

Pdzly committed:
Added Personmapper
and RoleMapper

jgrim commented on src/main/java/com/sublinks/sublinksapi/api/sublinks/v1/person/mappers/PersonMapper.java:
```diff
@@ -0,0 +1,58 @@
+package com.sublinks.sublinksapi.api.sublinks.v1.person.mappers;
+
+import com.sublinks.sublinksapi.api.lemmy.v3.utils.DateUtils;
```
Try to avoid cross-dependencies between Lemmy API and Sublinks API. There is a possibility that we'll drop Lemmy API support at some point.

jgrim commented on src/main/java/com/sublinks/sublinksapi/api/sublinks/v1/common/enums/DateSort.java:
```diff
@@ -0,0 +1,6 @@
+package com.sublinks.sublinksapi.api.sublinks.v1.common.enums;
+
+public enum DateSort {
```
Could this be more generic than just Date sort?

Pdzly committed:
Remove unnecessary blank line in PersonMapper

This commit removes an unnecessary blank line in the PersonMapper class. The removal cleans up the code and improves its readability. This change adheres to the defined code style guide.

Pdzly committed:
Merge branch 'refs/heads/feature/353-swap-to-postgres' into issue/119-native-api

# Conflicts:
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/site/controllers/SiteController.java

Pdzly committed:
Refactor PersonService, implement JWT auth and person registration

The code refactor involved updating PersonService's relevant methods and adjusting their arguments and properties. Alongside this, JSON Web Token (JWT) authentication has been implemented to secure APIs. A new process for person registration has also been introduced, which now includes email verification and handling different registration modes. New components such as JwtFilter, JwtUtil and JwtPerson have been included for managing JWTs, and a new timeline endpoint for default controller is also introduced.

jgrim commented on src/test/java/com/sublinks/sublinksapi/language/services/LanguageServiceUnitTests.java:
```diff
@@ -73,6 +73,6 @@ void givenCollectionOfLanguageIds_whenLanguageIdsToEntity_thenReturnListOfLangua
 
     assertEquals(2, languages.size(),
-        "Number of languages instances returned did not match expected");
+        "Number of languagesKeys instances returned did not match expected");
```
I liked this message as it was before. Why the change?

Pdzly replied:
a regex fail

jgrim commented on src/main/java/com/sublinks/sublinksapi/person/entities/PersonRegistrationApplication.java:
```diff
@@ -95,6 +94,7 @@ public final boolean equals(Object o) {
 
     return this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer()
-        .getPersistentClass().hashCode() : getClass().hashCode();
+        .getPersistentClass()
+        .hashCode() : getClass().hashCode();
```
Unnecessary, formatting change.

Pdzly replied:
It is within our .editorconfig

jgrim replied:
I just mean it's a change to formatting unrelated to the issue.

Pdzly replied:
We should run the editor config on every file in some time.

jgrim commented on src/main/java/com/sublinks/sublinksapi/person/entities/Person.java:
```diff
@@ -162,34 +162,31 @@ public class Person implements UserDetails, Principal {
   private ListingType defaultListingType;
 
-  @Column(nullable = false, name = "default_sort_type")
+  @Column(nullable = true, name = "default_sort_type")
```
I don't believe this should be nullable.

Pdzly replied:
then why it is in the sql?

jgrim replied:
Not sure, I guess it's okay to be nullable but we'll have to ensure the code accounts for that.

Pdzly replied:
It is, as it is already not null. So idk why it was null on the db migration.

jgrim commented on src/main/java/com/sublinks/sublinksapi/person/entities/Person.java:
```diff
@@ -162,34 +162,31 @@ public class Person implements UserDetails, Principal {
   private String defaultTheme;
 
-  @Column(nullable = false, name = "default_listing_type")
+  @Column(nullable = true, name = "default_listing_type")
```
I don't believe this should be nullable.

Pdzly replied:
^

jgrim commented on src/main/java/com/sublinks/sublinksapi/person/entities/Person.java:
```diff
@@ -162,34 +162,31 @@ public class Person implements UserDetails, Principal {
   private String interfaceLanguage;
 
-  @Column(nullable = false, name = "default_theme")
+  @Column(nullable = true, name = "default_theme")
```
I don't believe this should be nullable.

jgrim commented on src/main/java/com/sublinks/sublinksapi/person/entities/Person.java:
```diff
@@ -162,34 +162,31 @@ public class Person implements UserDetails, Principal {
   private String biography;
 
-  @Column(nullable = false, name = "interface_language")
+  @Column(nullable = true, name = "interface_language")
```
I don't believe this should be nullable.

jgrim commented on src/main/java/com/sublinks/sublinksapi/language/services/LanguageService.java:
```diff
@@ -34,4 +34,11 @@ public List<Language> languageIdsToEntity(final Collection<String> languageIds)
     }
     return languages;
   }
+
+  public List<Language> instanceLanguages(final Instance instance) {
+
+    final List<Language> languages = new ArrayList<>();
+    languages.addAll(instance.getLanguages());
+    return languages;
+  }
```
This method makes zero sense to me. It's taking a list and making a list from it, right?

jgrim commented on src/main/java/com/sublinks/sublinksapi/language/entities/Language.java:
```diff
@@ -27,7 +27,7 @@
 @Builder
 @Entity
-@Table(name = "languages")
+@Table(name = "languagesKeys")
```
Please, return the name to "languages"

jgrim commented on src/main/java/com/sublinks/sublinksapi/api/sublinks/v1/privatemessage/mappers/PrivateMessageMapper.java:
```diff
@@ -0,0 +1,27 @@
+package com.sublinks.sublinksapi.api.sublinks.v1.privatemessage.mappers;https://github.com/sublinks/sublinks-docker
```
```suggestion
package com.sublinks.sublinksapi.api.sublinks.v1.privatemessage.mappers;
```

jgrim reviewed (dismissed):
Please, add javadoc to classes and methods.

Pdzly committed:
Refactor codebase to switch primitive boolean to Boolean and expand API functionality

Made changes throughout the project by replacing primitive boolean with its Wrapper class, Boolean. This allows for null values, which can represent 'undefined' status. In addition, introduced new community-related endpoints and services, for enhanced community details management. Also, new SortType and ListingType enumerations were added, to enrich the sorting and listing options. Implemented relevant conversions and mapping for new types.

Pdzly committed:
Fixed a bug

Pdzly committed:
Fixed a bug

Pdzly committed:
Add find methods in repositories and enhance community service

Added method to find Instance by domain in the InstanceRepository and find methods for Community in the CommunityRepository. Enhanced SublinksCommunityService to handle community updates, deletion and removal, and validate unique titleSlugs upon creation. Also introduced ActorIdUtils for handling actorIDs.

Pdzly committed:
Implement comment service logic and enhance authorization

Added SublinksCommentService for handling comment creation with proper validations and exception handling. Adjusted RoleAuthorizingService by introducing helper methods to check a person's permission status. Updated CommunityModActionsController with refined exception handling logic. Added necessary find methods in PostRepository and CommentRepository. Introduced new model classes for comments.

Pdzly committed:
Refactor to use Optional in CommunityRepository methods

CommunityRepository has been refactored to use Optional for the return type of findCommunityByTitleSlug method to better handle absence of value situations. Additionally, logic has been added in createCommunity and updateCommunity methods to check user permissions for these operations. The updateCommunity method has also been simplified for better readability and efficiency.

jgrim commented on src/main/java/com/sublinks/sublinksapi/api/sublinks/v1/community/services/SublinksCommunityService.java:
```diff
@@ -54,24 +60,21 @@ public CommunityResponse createCommunity(CreateCommunity createCommunity, Person
+    if (domain != null && domain.equals(localInstanceContext.instance().getDomain())) {
+      key = ActorIdUtils.getActorId(key);
+    }
+    Optional<Community> foundCommunity = communityRepository.findCommunityByTitleSlug(key);
```
Instead of loading the entire object you can simply check if it exists for better performance. Checkout https://www.baeldung.com/spring-data-exists-query

Pdzly replied:
It is required further down.

Pdzly committed:
Refactor authentication and add language support to comments

This commit renames classes related to JWT authentication for coherence and better readability. It also includes a new column in the Language entity to handle language codes. Furthermore, a validation has been added to check language availability while creating comments. Minor syntax changes are also performed to improve readability and comply with updated editor settings. This refactoring ensures better code organization and enhances functionality by accepting comments in different languages.

Pdzly committed:
Merge branch 'refs/heads/main' into issue/119-native-api

# Conflicts:
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/community/controllers/CommunityModActionsController.java
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/community/controllers/CommunityOwnerController.java
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/post/controllers/PostController.java
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/user/controllers/UserAuthController.java
#	src/main/java/com/sublinks/sublinksapi/authorization/services/RoleAuthorizingService.java
#	src/main/java/com/sublinks/sublinksapi/person/repositories/PersonRepository.java
#	src/main/java/com/sublinks/sublinksapi/person/services/PersonService.java
#	src/main/java/com/sublinks/sublinksapi/person/services/UserDataService.java

Pdzly committed:
Refactor code for readability and consistency

Removed irrelevant code block in PersonService and improved formatting across various methods in CommunityModActionsController and PostController to improve readability and maintain consistency. Arranged method calls and parameters to make the code cleaner and easier to understand.

Pdzly committed:
.

Pdzly committed:
Refactor authentication and update APIs

The authentication config has been refactored and the APIs related to JWT filter, person, and community have been updated. The changes include filename changes, code reformatting, function renaming, and improved access control checks.

Pdzly committed:
Refactor controller classes and improve security

This commit refactors several controller classes for better code organization and consistency. Classes were renamed, and the RestController annotation was added to each controller. Additionally, the implementation of JWT filter was updated to improve authentication security. Improvements to the build file were also made, adding additional dependencies.

Pdzly committed:
Renamed Enum 'ListingType' to 'SublinksListingType' and updated references

The enum 'ListingType' was renamed to 'SublinksListingType' to be more specific. All references in the code to the old 'ListingType' has been updated to 'SublinksListingType'. Additionally, the boolean 'showNsfw' in IndexCommunity class was updated to not use Optional.

Pdzly committed:
Made the community controller work

Pdzly commented on src/main/java/com/sublinks/sublinksapi/person/services/PersonService.java:
```diff
@@ -230,10 +224,8 @@ public void createPerson(final Person person) {
-        .person(person)
-        .build());
+    person.setLinkPersonInstance(LinkPersonInstance.builder().instance(
+        localInstanceContext.instance()).person(person).build());
```
the editor config was unloaded somehow.

Pdzly committed:
Merge branch 'main' into issue/119-native-api

Pdzly committed:
Fixed some more

Pdzly committed:
Add CommunityView and CommunityAggregates models

New models, CommunityView and CommunityAggregates, were introduced to provide comprehensive information about communities. The SublinksCommunityController was also updated to use these models, thereby improving the details provided when fetching a community or list of communities.

Pdzly committed:
Fixed a hickup

Pdzly committed:
Renamed some attributes

Pdzly committed:
Renamed some attributes

Pdzly committed:
Revert "Fixed a hickup"

This reverts commit 05a030bc70ed10a859809aa0af00645149b09496.

Pdzly committed:
Revert "Renamed some attributes"

This reverts commit a76268e1

Pdzly committed:
Revert "Add CommunityView and CommunityAggregates models"

This reverts commit 0d47e472c494a7fa404ce7ef0871f022351a74e3.

Pdzly committed:
Merge remote-tracking branch 'origin/issue/119-native-api' into issue/119-native-api

Pdzly committed:
Revert "Renamed some attributes"

This reverts commit 171f0b7a2272e145cf22e68ded4cc7a35bdc6987.

Pdzly committed:
Revert "Renamed some attributes"

This reverts commit a76268e1109ce7c7521f88ad88e345239bf93118.

Pdzly committed:
Revert "Fixed a hickup"

This reverts commit 05a030bc70ed10a859809aa0af00645149b09496.

Pdzly committed:
Revert "Add CommunityView and CommunityAggregates models"

This reverts commit 0d47e472c494a7fa404ce7ef0871f022351a74e3.

Pdzly committed:
Renamed some attributes

Pdzly committed:
Add community moderation features

Designed and implemented moderation functionalities in the Sublinks community. This includes obtaining community moderators, adding and deleting moderators, and the capacity to ban users. Fixed the API paths for some services and introduced new types and mapper classes to align with the new moderation features.

urda commented:
My bad I'm not sure what I did to trip the few issues I impacted today. Sorry gang.

jgrim committed:
Merge branch 'main' into issue/119-native-api

Pdzly committed:
Implement user ban and unban functionality

The commit introduces methods to ban and unban users from a community. To implement this change, a new import for PersonRepository was added to the SublinksCommunityService class. New endpoints for "ban" and "unban" were added to the SublinksCommunityModerationController. The findOneByName method in PersonRepository was replaced with findOneByNameIgnoreCase to ensure the case insensitivity of username in ban/unban requests. In addition, The LinkPersonCommunityService class was updated to handle the removal of any user link in case of banning.

Pdzly committed:
Add CRUD features for communities and initial post functionality

In this update, several features were added to the community services including the ability to create, update, and retrieve communities. In addition, functionality to manage moderation including getting community moderators and banning/unbanning persons was introduced. Also introduced initial pieces for posting functionality with new models and mapping methods. Necessary documentation for these features was added in addition to minor formatting changes.

Pdzly committed:
Add community and comment moderation features

Introduced delete and remove functionality to both community and comment sections of the platform. Changed tag names in community controllers for better clarity. Updated Community and Comment services to support new moderation features. Also, made minor adjustments to PostMapper and RolePermissionCommentTypes for better code organization.

Pdzly committed:
Refactor UserController to improve person update logic

This commit breaks down the original combined setters and condition checks in UserController when updating the Person object. This makes the code more readable and ensures that each field is properly checked before assignment. The code is now more maintainable, as each operation is clearly isolated.

Pdzly committed:
Update comment controller to return a list of responses

The change have been made in the SublinksCommentController class, specifically in the index method, to return a list of CommentResponse objects instead of a single object. This is a more efficient way to handle multiple comments.

Pdzly committed:
Enhance comment and community features

The commit adds the ability to view, create and edit comments. Updates were made to the community service to provide functionality for retrieving a list of communities based on certain criteria, creating a comment, and updating comments. The addition of authorization checks was also made in several places to ensure only authorized users can perform certain operations. Additionally, functionalities were created to interact with community moderators and community aggregation.

Pdzly committed:
Enhance ban logic and add community remove feature

This commit significantly enhances the ban logic and introduces the community remove feature in the community moderation functionality. Specifically, the ban logic has been updated to allow both banning and unbanning of users. The commit adds a remove method that enables moderators to remove communities based on roles and permissions for both admin and moderator roles have been updated accordingly. As part of this enhancement, several record models related to community moderation have been updated or created.

Pdzly committed:
Enhance ban logic and add community remove feature

This commit significantly enhances the ban logic and introduces the community remove feature in the community moderation functionality. Specifically, the ban logic has been updated to allow both banning and unbanning of users. The commit adds a remove method that enables moderators to remove communities based on roles and permissions for both admin and moderator roles have been updated accordingly. As part of this enhancement, several record models related to community moderation have been updated or created.

Pdzly committed:
Merge remote-tracking branch 'origin/issue/119-native-api' into issue/119-native-api

Pdzly committed:
Merge branch 'main' into issue/119-native-api

Signed-off-by: Pdzly <34943569+Pdzly@users.noreply.github.com>

Pdzly committed:
Merge branch 'refs/heads/main' into issue/119-native-api

Pdzly committed:
Add comment pinning feature and refactor HTTP methods

The commit includes the implementation of a new comment pinning feature in the comment service and controller, which will allow moderators to pin and unpin comments. It also includes changes in HTTP methods from 'GET' to 'POST' in several places to align with REST best practices for methods that change the state of the data. Some formatting adjustments for improve code readability are also included.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add comment pinning functionality and refactor HTTP methods

This commit introduces the ability for moderators to pin and unpin comments through the new functionality in the comment service and controller. It also replaces several 'GET' HTTP methods with 'POST' for compliance with REST standards on state-changing operations. Additionally, some code formatting was adjusted for better readability.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly commented on src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/authentication/config/SecurityConfig.java:
```diff
@@ -32,16 +34,15 @@
-        )
-        .addFilterBefore(jwtFilter, UsernamePasswordAuthenticationFilter.class);
+    http.addFilterBefore(lemmyJwtFilter, UsernamePasswordAuthenticationFilter.class);
+    http.csrf(AbstractHttpConfigurer::disable)
```
@jgrim i am not sure about this, but why do we disable csrf?

Pdzly committed:
Refactor application and extend functionality

This commit updates and enhances the application in multiple ways. It includes refactoring classes that have been moved to more appropriate packages and renaming some of them for better understanding. Furthermore, new functionalities have been introduced such as implementing a service to ban a user and a service to retrieve a list of banned users. The code also contains improved error handling and minor modifications for improving code quality.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Merge branch 'main' into issue/119-native-api

Signed-off-by: Rooki <34943569+Pdzly@users.noreply.github.com>

Pdzly committed:
Refactor PersonAggregationController and add Auth checks

Changes have been made to refactor the `SublinksPersonAggregationController`. The controller now makes use of the `PersonAggregateResponse` and `PersonIdentity` models. In addition, authentication checks have been added to ensure a user has the correct permissions before accessing certain data.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Enhanced authorization checks and response models

The commit introduced more detailed authorization checks in various parts of the application. It further enriched the response models for better client-side readability. Moreover, annotation responses have been updated in multiple controllers to improve the OpenAPI documentation.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add search functionality for instances and enhance APIs

Introduced search functionality in the Instance Repository to retrieve instances based on domain or description. Enhanced various controllers for better data handling and error management. Updated the InstanceConfig and Person Repositories for increased efficiency. Enhanced the application's mapping capabilities by introducing a new SublinksInstanceMapper. Added RegistrationMode and ListingType enums for better data management in instances.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
LINT

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Extend Search API to support communities, posts, and comments

In this update, the Search API has been expanded to include not only person-related responses but also community, post, and comment responses. A new service, SublinksSearchService, has been added, along with several model updates to support these features. Additionally, code adjustments have been made to adapt the parameter renaming of 'limit' to 'perPage'.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor code structure and add search functionality

Code has been refactored for better readability and new methods have been added to the Builder class in PostSearchQueryService.java for filtering NSFW posts and to perform a search. Improvements to code structuring have been made across multiple files to enhance readability. Additionally, search functionality parameters have been added to the IndexCommunity.java file.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Optimize search functionality and improve code structure

Refactored code for increased readability, and added new methods for NSFW post filtering and search functionality in PostSearchQueryService.java. Enhanced overall code structuring across various files, and incorporated search function parameters within IndexCommunity.java.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor PostReport and PostLike repositories

This commit refactors the PostReportRepositoryImpl and PostLikeRepositoryImpl classes, aiming to improve the code readability and maintainability. Changes include updating the method argument structure, introducing the use of the applyPagination utility function to manage pagination in queries, and reformatting the code for increased readability.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add show and create methods to SublinksPostService

Several new methods and imports have been added to the SublinksPostService. The `show` method allows to find and display a specific post based on a key and a person, performing various checks to ensure post visibility. The `create` method executes complex functionality to create a new post. Furthermore, CreatePost model has also been updated with new parameters to support these changes.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Merge branch 'refs/heads/main' into issue/119-native-api

# Conflicts:
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/post/controllers/PostController.java
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/user/controllers/UserController.java
#	src/main/java/com/sublinks/sublinksapi/authorization/services/InitialRoleSetupService.java
#	src/main/java/com/sublinks/sublinksapi/comment/repositories/CommentRepository.java
#	src/main/java/com/sublinks/sublinksapi/post/repositories/PostRepositoryImpl.java
#	src/main/java/com/sublinks/sublinksapi/post/services/PostSearchQueryService.java

Pdzly committed:
Enhance Post functionalities with language support and slur filter

This commit includes enhancements to the 'Post' object. Language restrictions are added, checks whether the post language is supported by the community or instance are implemented. A Slur filter has been integrated to handle exception scenarios and it can censor text in post bodies. Additionally, metadata from the linked website, if a link is provided in the post, is fetched and stored.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Implement comment aggregate API and extend post functionalities

This update introduces the comment aggregate API, providing insights on aggregate data like upvotes, downvotes, hot rank and reply count per comment. Furthermore, it broadens the post functionalities by introducing language restrictions and enforcing to check if the chosen language is supported by the respective community or instance. The slur filter integration helps in text censorship within post bodies, and links within posts now have metadata fetching and storing ability.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Implement comment aggregate API and extend post functionalities

This update introduces the comment aggregate API, providing insights on aggregate data like upvotes, downvotes, hot rank and reply count per comment. Furthermore, it broadens the post functionalities by introducing language restrictions and enforcing to check if the chosen language is supported by the respective community or instance. The slur filter integration helps in text censorship within post bodies, and links within posts now have metadata fetching and storing ability.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Merge branch 'main' into issue/119-native-api

Pdzly committed:
Remove unnecessary comment from SQL migration script

The diff shows a particular comment section being removed from a SQL migration script. The comment is no longer relevant or necessary, thereby enhancing the clarity and readability of the code.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Merge branch 'refs/heads/main' into issue/119-native-api

# Conflicts:
#	src/main/java/com/sublinks/sublinksapi/comment/models/CommentSearchCriteria.java

Pdzly committed:
Add announcement functionality to Sublinks API

This update added announcement handling functionality to Sublinks API. It includes creation of new model classes, updating service layers with query methods, and CRUD operations in the controller. It also added local people filtering to the `SublinksPersonService`.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add CRUD operations and models for announcements

The commit introduces full CRUD functionality for announcements in the Sublinks API. It includes creation of new Announcement models with related update and removal forms. Implementation includes service updates, repository additions, and corresponding changes in the Announcement Controller.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add 'active' status and 'creator' field to announcements

Added an 'active' status field to announcements, allowing for them to be toggled on and off. The 'creator' field was also added to link announcements to the user who created them. Renamed a comment mapper class, added a few new mapper classes and adjusted controller methods to accommodate these changes. Meanwhile, made adjustments to database queries and schemas to align with these updates.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor permissions checks in various services

Moved permissions checks from controller to service layer to ensure better encapsulation. This change affects SublinksCommunityController, SublinksCommentAggregateController, and SublinksCommentModerationController. An ADMIN_UPDATE_COMMUNITY permission was also added in RolePermissionCommunityTypes. This improves code maintainability and readability.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add functionality for handling user metadata

Introduced new endpoints in the SublinksPersonSessionController to allow managing user metadata including getting, invalidating, and deleting metadata. Specific permissions have been added to govern these actions. Two new models, PersonSessionData and PersonSessionDataResponse, have been included to manage data transfer. Enhancements are made in services for consistent permission checks and handling new operations.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Merge branch 'refs/heads/main' into issue/119-native-api

# Conflicts:
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/authentication/config/SecurityConfig.java

Pdzly committed:
Refactor security configuration in SecurityConfig.java

This commit simplifies the code in the SecurityConfig.java file for the security configuration of the Lemmy V3 API. It removes some duplication in the code related to session management and the authorization of HTTP requests, and changes the order of the `addFilterBefore` method to execute after session management.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Rearrange JWT filter addition in SecurityConfig

The JWT filter addition is rearranged in the SecurityConfig. Previously, it was added before configuring CSRF, securityMatcher, authorization, and sessionManagement. Now, it is added after configuring all these. This change ensures that the JWT filter gets executed at the correct point in the filter chain.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor SublinksPersonMetaDataMapper class

The commit removes unused methods and imports from the SublinksPersonMetaDataMapper class. The 'updatedAt' mapping source has also been changed from 'metadata.updatedAt' to 'metadata.lastUsedAt'. Besides, the 'mapPersonKey' method and associated Person import have been removed since they are no longer being used.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor code and enhance User Data Management capabilities

Significant improvements have been added to the management of user data and user sessions. The changes include the correction of certain URL Utilities, updates to the pagination functionalities, and enhancements to the handling of user metadata and validation mechanisms. Additionally, there are changes to the roles functionality, focusing on improving permissions handling, and also improvements on the structure of the API responses, and creation of essential utilities for person key handling.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor UserDataService to simplify metadata updates

The UserDataService has been refactored for simpler active user data retrieval and updating. The fetching process has been simplified by removing the need to fetch data using person and IP address together. Also, the metadata update procedure has been improved; it no longer needs to instantiate a new metadata every time there's a change, but instead, modifies the existing metadata instance.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Reorder import statements in UserDataService

The import statements in the UserDataService class have been reordered. This change improves readability and adheres to code style guidelines. No functional changes have been made to the code.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add authorization to instance config actions

The methods for showing and updating instance configurations now require the user's identity. The server will throw unauthorized responses when requested by unauthorized people. Permission types for reading and updating instances were added to the RolePermissionInstanceTypes enumeration.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor moderation models and update relevant methods

Renamed moderation models for consistency and clarity, and updated all relevant methods in service and controller classes accordingly. Two new models for Post moderation were added. Additionally, fixed logic in checking delete post permission in SublinksPostService.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add post aggregate functionality

Renamed Aggregate model classes to unify nam

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Update moderation actions from "remove" to "pin"

The code alters several parts where a moderator could remove certain entities. The action has now been changed to pinning the entities instead. This includes changes in comments, posts, and community moderation. Additional adjustments have been made to ensure correct behaviour of the new pin functionality.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add purge community operation and update delete action summary

A new operation to purge a community was added to the Sublinks Community Moderation Controller. This operation, however, is still not implemented and will be completed in the future. The summary text for the delete operation in the Sublinks Community Controller was changed from 'Purge an community' to 'Delete a community'.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add post pinning, community pinning, and post purging methods

New functionalities added to SublinksPostModerationController: pinning a post, pinning a post in a community, and purging a post. The implementation for purging a post is not complete yet, but will be in future iterations. Additionally, the description for the Post Moderation API is updated. Existing functionality in the SublinksPostService is adjusted to allow the "featured" status of a post to be toggled during the pinning process.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Update permissions and search functionalities in API

Enhancements were made to various API functionalities, including the addition of Instance Search and revised Private Message features. An array of new dependencies and properties were defined for instances and private message search criteria. SearchScopes now encompass posts, comments, communities, and instances. The changes aim to improve the user's search experience across categories and streamline the process of private message handling.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Remove "purge" functionality and cleanup controllers

The "purge" functionality has been removed from the SublinksCommunityModerationController. Additionally, unneeded component and dependencies were removed from the SublinksCommunityController, and the constructor in SublinksPrivatemessageModeratorController was replaced with @AllArgsConstructor for brevity.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Change request mapping in SublinksPostAggregateController

The request mapping for the SublinksPostAggregateController class has been updated. Previously, it was incorrectly mapped to "api/v1/comment/{key}/aggregate". It has been corrected to "api/v1/post/{key}/aggregate".

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Merge branch 'refs/heads/main' into issue/119-native-api

# Conflicts:
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/authentication/config/SecurityConfig.java
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/comment/controllers/CommentModActionsController.java
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/community/controllers/CommunityOwnerController.java
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/privatemessage/controllers/PrivateMessageController.java
#	src/main/java/com/sublinks/sublinksapi/comment/repositories/CommentRepositoryImpl.java
#	src/main/java/com/sublinks/sublinksapi/person/services/UserDataService.java
#	src/main/resources/db/migration/V20231006__Create_foreign_key_references.sql

Pdzly committed:
Enhance role management and authorization handling.

Removed `Role` model and replaced it with new event-based role management. Introduced publishers and events for role creation, updates, and deletion, and updated controllers and mappers accordingly. Additionally, simplified authorization error messages for consistency.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Update role management and ACL services for role expiration

Modified SQL scripts to refine unique index for roles table. Updated the role service to handle expiration dates in role creation and updates. Introduced ACL checks in post-related services to enhance authorization, and adjusted controller methods to align with these changes.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor RequestResponse imports and enhance RolesController

Renamed RequestResponse.java to its appropriate package path to improve code organization. Additionally, updated RolesController to accept RequestBody and manage roles more effectively by adding implementations for person-based CRUD operations on roles.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Merge branch 'refs/heads/main' into issue/119-native-api

# Conflicts:
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/community/controllers/CommunityModActionsController.java
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/post/controllers/PostModActionsController.java
#	src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/user/services/LemmyPersonMentionService.java
#	src/main/java/com/sublinks/sublinksapi/comment/repositories/CommentRepositoryImpl.java
#	src/main/java/com/sublinks/sublinksapi/person/services/LinkPersonCommunityService.java
#	src/main/java/com/sublinks/sublinksapi/person/services/LinkPersonPostService.java
#	src/main/java/com/sublinks/sublinksapi/post/services/PostService.java

Pdzly committed:
Add inheritance support to roles and enhance role mapping.

Added an "inherits_from" column to the acl_roles table with a foreign key reference. Updated role mapping services to handle inherited permissions and refactored related mappers and services to integrate with the updated role structure.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Enable Hikari pool suspension and update transactional annotations

Added properties for Hikari connection pool suspension and PostgreSQL dialect to `application.properties`. Replaced `jakarta.transaction.Transactional` with `org.springframework.transaction.annotation.Transactional` and updated method implementations in `InitialRoleSetupService`.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor role saving to use saveAndFlush

Changed repository calls to saveAndFlush for immediate persistence. Updated savePermissions method to return void and improved role initialization with empty permissions set. This change ensures that roles and permissions are persistently stored in the database immediately.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor role and permissions mapping for improved handling

Simplify the SublinksRoleMapper by removing the injection of unused dependencies and use lambda expressions mapping permissions directly. Update InitialRoleSetupService to use appropriate transaction handling and entity merging for improved data consistency. Adjust Role and RolePermission entities to correctly manage relationships with cascade operations for easier maintenance of role-based permissions.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Enhance mapper configurations and simplify role permission logic

Added unmapped target policy to SublinksPersonMapper to ignore mapping issues. Removed unused import and simplified permission saving logic in InitialRoleSetupService. Refactored SublinksRoleMapper by removing dependency on RolePermissionService and added methods for mapping inherited permissions.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor role service pagination and formatting of controller methods

Reorganized pagination logic in SublinksRoleService for better readability and consistency. Improved code formatting in AdminController by aligning opening braces for better style compliance and readability.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add missing newline for better readability

Added a newline between import statements in AdminController.java. This enhances code readability and conforms to standard formatting practices. No functional changes were made.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add purge post functionality to moderation

Implemented a new purge feature with corresponding service method and controller endpoint to handle requests. Added the PurgePost model to represent the request payload and updated ACL service to check permissions.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Implement purge post placeholder in SublinksPostService

Added a placeholder for the purge post functionality in SublinksPostService. This includes a TODO comment to mark where the implementation should occur and sets the response to indicate the feature is not yet implemented.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor method parameter name in purge method

Updated the parameter name from `removePostForm` to `purgePostForm` in the `purge` method for better consistency and clarity. This change improves code readability and maintains alignment with naming conventions used throughout the class.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add @NonNull annotations and comprehensive JavaDocs

Enhanced the codebase with @NonNull annotations to prevent null values and added detailed JavaDocs for better method documentation. This improves code robustness and readability by clearly specifying non-null contract and providing usage guidelines for methods.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Adjust pagination to start from the first page.

Previously, the pagination could start at an invalid state due to using zero as the lower bound. This change ensures that pagination correctly starts at page 1 instead of 0.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add SublinksCommentSortTypeMapper and refactor sort type logic

Introduced SublinksCommentSortTypeMapper to map SortType to CommentSortType. Refactored SublinksCommentService for improved sortType and listingType handling, ensuring null checks and default value assignments. Removed redundant imports in the affected classes.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor comment entities and service methods

Updated the Comment entity to have a nullable searchVector, modified service logic for saving comments, and added toBuilder for CommentResponse. Refactored SublinksRoleMapper usage and improved formatting consistency in service classes.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Remove replies field from CommentResponse model

Removed the `replies` field and associated annotations from the `CommentResponse` model. Updated dependent mappers and services to reflect this change. This simplifies the comment structure and reduces unnecessary data handling.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Remove @Builder annotation from CommentResponse

The @Builder annotation was removed from the CommentResponse record. This change likely addresses redundancy or issues caused by the builder pattern in this specific context.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add instance data to PersonResponse and improve search handling

Added InstanceResponse to PersonResponse model and integrated ACL service in person search for better permissions handling. Refactored pagination and search methods, optimizing query execution, and cleaned up unused imports.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor comment handling and clean up unused dependencies

Implemented a tree structure for handling comment replies, enabling nested comment representation. Removed unused dependencies and redundant fields in various controllers and services for better code maintainability. Also adjusted mappers to accommodate new replies structure, improving overall data handling.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Clarify comment handling in buildHierarchy method

Updated the JavaDoc to specify that comments without replies or missing parents are placed at the end of the list. This documentation update ensures developers understand the behavior when organizing comments into a hierarchy.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Fix incomplete comment in SublinksCommentService.java

Corrected the documentation comment in the build comment tree method. This ensures the explanation for the comment reply structure is clear and complete, aiding future maintenance and understanding.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Fix typos in comment structuring logic documentation.

Corrected description to clarify handling of comments with no parent or missing parent in the list. This ensures better understanding of how orphaned comments are treated in the tree building process.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Refactor pagination logic and improve SQL query

Refactored pagination handling into utility methods in `PaginationUtils` for consistency and reusability. Updated SQL query in `InstanceRepository` to correct syntax and improve domain search functionality. Enhanced readability by simplifying instance response conversion logic in `SublinksInstanceService`.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Standardize API tags and refactor unauthorized exception

Standardized the naming of API tags across multiple controllers for clarity. Refactored exception handling to use the `orThrowUnauthorized` method, reducing redundancy in authorization checks.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add Swagger annotations to community models

Introduced Swagger annotations for `CreateCommunity`, `DeleteCommunity`, and `IndexCommunity` models. This improves API documentation by providing detailed descriptions and default values for relevant fields. These changes enhance developer understanding and API usability.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add Swagger annotations to CommentResponse and UpdateComment

Added Swagger annotations to fields in `CommentResponse` and `UpdateComment` for improved API documentation clarity. Annotations provide thorough descriptions and specify required fields, enhancing the maintainability and user comprehension of the API.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add Swagger annotations and rename model class

Added Swagger annotations to enhance API documentation across multiple community and comment models. Renamed `AggregateCommentResponse` to `CommentAggregateResponse` for consistency and updated the relevant references in the service and controller files. Also included title update handling in the Community service logic.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Fix indentation in SublinksCommentAggregateController

Corrected the indentation in the aggregate method to improve code readability. This change ensures consistent formatting and maintains coding standards.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Merge branch 'main' into issue/119-native-api

Pdzly committed:
Remove RolePermissionService from SublinksCommentModerationController

Refactored the SublinksCommentModerationController by removing the unused RolePermissionService dependency. This cleanup simplifies the code and removes redundant imports, enhancing maintainability.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add token handling to session management

Introduced token field and related methods for handling user sessions. Added endpoints for managing session invalidation and logout, enhancing session control functionalities.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Format code for consistency in SublinksAnnouncementService

Standardized formatting in the SublinksAnnouncementService class to enhance readability and maintain consistency. Primarily adjusted spacing in conditional statements and response status exception lambdas.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add JWT token to SublinksJwtPerson object

Included the JWT token in the SublinksJwtPerson object to ensure the token is accessible during authentication. This improvement enhances the security context by embedding the token with user authorities.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
LINT

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Change submodule URL for 'docker' repository

Updated the URL of the 'docker' submodule to use HTTPS instead of SSH, pointing to a new repository. This change affects the '.gitmodules' file to reference the new repository location.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Change submodule URL for 'docker' repository

Updated the URL of the 'docker' submodule to use HTTPS instead of SSH, pointing to a new repository. This change affects the '.gitmodules' file to reference the new repository location.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Add password encryption in SublinksPersonService

The password field from createPersonForm is now included when building a new Person object. This ensures the password gets encrypted as part of the person creation process.

Signed-off-by: rooki <rooki@rooki.xyz>

Pdzly committed:
Handle missing tokens in JwtFilters

Added checks to return "invalid_token" error if JWT validation fails in both JwtFilter and SublinksJwtFilter classes. Simplified AddAdmin model by adjusting constructor format.

Signed-off-by: rooki <rooki@rooki.xyz>
